u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -215,4 +291,65 @@ public static AuthenticateResponse decodeAuthenticateResponse(byte[] data) throw
     .put(challengeSha256);
     return signedData;
   }
+  
+  public static byte[] encodeTransferAccessMessageSignedBytesForAuthenticationKey(
+      byte sequenceNumber, byte[] newUserPublicKey, byte[] applicationSha256,
+      X509Certificate attestationCertificate) throws U2FException {
+
+    byte[] attestationCertificateBytes;
+    try {
+      attestationCertificateBytes = attestationCertificate.getEncoded();
+    } catch (CertificateEncodingException e) {
+      throw new U2FException("Error when encoding attestation certificate.", e);
+    }
+
+    byte[] signedData = new byte[1 + newUserPublicKey.length + applicationSha256.length
+        + attestationCertificateBytes.length];
+    
+    ByteBuffer.wrap(signedData)
```

Comment:
It's just hard to make sure the size is correct when you pre-allocate.  It's also hard to maintain.  I strongly suggest not pre-allocating.